src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -729,4 +807,254 @@ public static function provideReplaceIgnoreCase()
+    }
+
+    /**
+     * @dataProvider provideStartsWithIgnoreCase
+     */
+    public function testStartsWithIgnoreCase(bool $expected, string $origin, $prefix)
+    {
+        $this->assertSame($expected, static::createFromString($origin)->ignoreCase()->startsWith($prefix));
+    }
+
+    public static function provideStartsWithIgnoreCase()
+    {
+        return [
+            [false, '', ''],
+            [false, '', 'foo'],
+            [false, 'foo', ''],
+            [false, 'foo', 'o'],
+            [false, "\nfoo", 'f'],
+            [true, 'foo', 'F'],
+            [true, 'FoO', 'foo'],
+            [true, 'foo', new BinaryString('F')],
+            [true, 'foo', new Utf8String('F')],
+            [true, 'foo', new GraphemeString('F')],
+            [true, 'foo', ['E', 'F', 'G']],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEndsWith
+     */
+    public function testEndsWith(bool $expected, string $origin, $suffix)
+    {
+        $this->assertSame($expected, static::createFromString($origin)->endsWith($suffix));
+    }
+
+    public static function provideEndsWith()
+    {
+        return [
+            [false, '', ''],
+            [false, '', 'foo'],
+            [false, 'foo', ''],
+            [false, 'foo', 'f'],
+            [false, 'foo', 'O'],
+            [false, "foo\n", 'o'],
+            [true, 'foo', 'o'],
+            [true, 'foo', 'foo'],
+            [true, 'foo', new BinaryString('o')],
+            [true, 'foo', new Utf8String('o')],
+            [true, 'foo', new GraphemeString('o')],
+            [true, 'foo', ['a', 'o', 'u']],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEndsWithIgnoreCase
+     */
+    public function testEndsWithIgnoreCase(bool $expected, string $origin, $suffix)
+    {
+        $this->assertSame($expected, static::createFromString($origin)->ignoreCase()->endsWith($suffix));
+    }
+
+    public static function provideEndsWithIgnoreCase()
+    {
+        return [
+            [false, '', ''],
+            [false, '', 'foo'],
+            [false, 'foo', ''],
+            [false, 'foo', 'f'],
+            [false, "foo\n", 'o'],
+            [true, 'foo', 'O'],
+            [true, 'Foo', 'foo'],
+            [true, 'foo', new BinaryString('O')],
+            [true, 'foo', new Utf8String('O')],
+            [true, 'foo', new GraphemeString('O')],
+            [true, 'foo', ['A', 'O', 'U']],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEnsureStart
+     */
+    public function testEnsureStart(string $expectedString, string $origin, $prefix)
+    {
+        $instance = static::createFromString($origin)->ensureStart($prefix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureStart()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['foo', 'foo', 'foo'],
+            ['foobar', 'foobar', 'foo'],
```

Comment:
this is failing, current result is `bar`

Reply:
fixed